ambry-store/src/main/java/com.github.ambry.store/StoreDescriptor.java

```diff
@@ -0,0 +1,125 @@
+   * @param dataDir the directory path to locate the Store Descriptor file
+   * @param storeId the StoreId of the store for which the StoreDescriptor has to be instantiated
+   * @throws IOException
+   */
+  public StoreDescriptor(String dataDir, String storeId) throws IOException {
+    this.storeId = storeId;
+    File storeDescriptorFile = new File(dataDir, StoreDescriptor.STORE_DESCRIPTOR);
+    if (storeDescriptorFile.exists()) {
+      DataInputStream dataInputStream = new DataInputStream(new FileInputStream(storeDescriptorFile));
+      StoreDescriptor storeDescriptor = StoreDescriptor.fromBytes(dataInputStream);
+      this.incarnationId = storeDescriptor.incarnationId;
+    } else {
+      this.incarnationId = UUID.randomUUID();
+      storeDescriptorFile.createNewFile();
+      DataOutputStream dataOutputStream = new DataOutputStream(new FileOutputStream(storeDescriptorFile));
+      dataOutputStream.write(toBytes());
+      dataOutputStream.close();
+    }
+  }
+
+  /**
+   * Instantiates the StoreDescriptor with the incarnationId for the given store
+   * @param storeId
+   * @param incarnationId
+   */
+  private StoreDescriptor(String storeId, UUID incarnationId) {
+    this.storeId = storeId;
+    this.incarnationId = incarnationId;
+  }
+
+  public String getStoreId() {
+    return storeId;
+  }
+
+  public UUID getIncarnationId() {
+    return incarnationId;
+  }
+
+  /**
+   * Deserialize the {@link StoreDescriptor} from the stream
+   * @param stream the stream containing the store descriptor
+   * @return the deserialized {@link StoreDescriptor}
+   * @throws IOException
+   */
+  static StoreDescriptor fromBytes(DataInputStream stream) throws IOException {
+    StoreDescriptor storeDescriptor;
+    short version = stream.readShort();
+    switch (version) {
+      case VERSION_0:
+        String storeId = Utils.readIntString(stream);
+        // read incarnationId
+        String incarnationId = Utils.readIntString(stream);
+        UUID incarnationIdUUID = UUID.fromString(incarnationId);
+        storeDescriptor = new StoreDescriptor(storeId, incarnationIdUUID);
+        break;
+      default:
+        throw new IllegalArgumentException("Unrecognized version in StoreDescriptor: " + version);
+    }
+    return storeDescriptor;
+  }
+
+  /**
+   * Returns the serialized form of this {@link StoreDescriptor}
+   * @return the serialized form of this {@link StoreDescriptor}
+   */
+  private byte[] toBytes() {
+    byte[] storeIdBytes = storeId.toString().getBytes();
+    byte[] incarnationIdBytes = incarnationId.toString().getBytes();
+    int size = VERSION_SIZE + STORE_ID_LENGTH_SIZE + storeIdBytes.length + INCARNATION_ID_LENGTH_SIZE +
+        incarnationIdBytes.length;
+    byte[] buf = new byte[size];
+    ByteBuffer bufWrap = ByteBuffer.wrap(buf);
+    // add version
+    bufWrap.putShort(VERSION_0);
+    bufWrap.putInt(storeIdBytes.length);
+    bufWrap.put(storeIdBytes);
```

Comment:
I don't think the `storeId` part is really thought through. Why do we need this? What do you do in the constructor if the passed in `storeId` is different from the one that is read in from the file?